sonar-project.properties

```diff
@@ -0,0 +1,32 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+sonar.organization=apache
+sonar.projectKey=apache_inbubator-daffodil
+
+# --- optional properties ---
+
+# defaults to project key
+#sonar.projectName=My project
+# defaults to 'not provided'
+#sonar.projectVersion=1.0
+
+# Path is relative to the sonar-project.properties file. Defaults to .
+sonar.modules=daffodil-cli,daffodil-core,daffodil-io,daffodil-japi,daffodil-lib,daffodil-macro-lib,daffodil-propgen,daffodil-runtime1,daffodil-runtime1-unparser,daffodil-sapi,daffodil-tdml-lib,daffodil-tdml-processor,daffodil-test,daffodil-test-ibm1,daffodil-udf
+sonar.sources=src/main
+sonar.tests=src/it,src/test
+sonar.java.binaries=target/**/classes
+sonar.java.test.binaries=target/**/test-classes
+sonar.java.libraries=../lib_managed/**/*.jar
```

Comment:
I guess maybe this is why you compile things? Does sonar cube do static analysis on jars/classes in addition to source?

Reply:
Yes, the binaries are required (according to their manula) But the libraries are optional, and it claims providing this gives it a fuller picture and not providing them might lead to imprecise data. I'll test locally to see how much of a different it makes towards results, but the compilation step is necessary for the scan to yield any results.